junebug/api.py

```diff
@@ -584,6 +556,97 @@ def delete_router_destination(self, request, router_id, destination_id):
 
         returnValue(response(request, 'destination deleted', {}))
 
+    @app.route(
+        '/routers/<string:router_id>/destinations/<string:destination_id>/messages/',  # noqa
+        methods=['POST'])
+    @json_body
+    @validate(
+        body_schema({
+            'type': 'object',
+            'properties': {
+                'to': {'type': 'string'},
+                'from': {'type': ['string', 'null']},
+                'group': {'type': ['string', 'null']},
+                'reply_to': {'type': 'string'},
+                'content': {'type': ['string', 'null']},
+                'event_url': {'type': 'string'},
+                'event_auth_token': {'type': 'string'},
+                'priority': {'type': 'string'},
+                'channel_data': {'type': 'object'},
+            },
+            'required': ['content'],
+            'additionalProperties': False,
+        }))
+    @inlineCallbacks
+    def send_destination_message(
+            self, request, body, router_id, destination_id):
+        '''Send an outbound (mobile terminated) message'''
+        router = yield Router.from_id(self, router_id)
+        router.get_destination(destination_id)
+        channel_id = router.router_worker.get_destination_channel(
+            destination_id)
+
+        msg = yield self.send_messsage_on_channel(channel_id, body)
+
+        returnValue(response(
+            request, 'message submitted', msg, code=http.CREATED))
+
+    @app.route(
+        '/routers/<string:router_id>/destinations/<string:destination_id>/messages/<string:message_id>',  # noqa
+        methods=['GET'])
+    @inlineCallbacks
+    def get_destination_message_status(
+            self, request, router_id, destination_id, message_id):
+
+        router = yield Router.from_id(self, router_id)
+        channel_id = router.router_worker.get_destination_channel(
+            destination_id)
```

Comment:
I'm not sure if this is possible in the `GET`.

Reply:
The message should be in the message store, so we should be able to load the body from the message store, and pass it to the function?

Reply:
To get the message from the store I need the channel id.

Reply:
Ah right. Well maybe we can send a message ID on the `GET`, and the message body on the `POST`? That way, on the `GET`, it can check each channel's message store until it finds the message, and the POST it already has the message body, so it knows where to route.